pkg/registries/ecr/ecr.go

```diff
@@ -84,7 +89,8 @@ func (e *ecr) Config() *types.Config {
 	if e.transport == nil {
 		return e.Registry.Config()
 	}
-	if err := e.transport.ensureValid(); err != nil {
+	ctx := context.Background()
```

Comment:
There is at least a couple of places, where the parent context could be passed to this function: `pkg/signatures/cosign_sig_fetcher` and `sensor/common/scan`.